netket/operator/_sumoperators.py

```diff
@@ -40,6 +40,21 @@ def __init__(
             coefficients: A coefficient for each ContinuousOperator object
             dtype: Data type of the matrix elements. Defaults to `np.float64`
         """
+        if len(operators) != len(coefficients):
+            raise AssertionError("Each operator needs a coefficient")
+
+        new_operators = []
+        new_coeffs = []
+        for op, c in zip(operators, coefficients):
+            if isinstance(op, SumOperator):
+                new_operators = new_operators + op._ops
+                new_coeffs = new_coeffs + list(c * jnp.array(op._pack_arguments()[0]))
+            else:
+                new_operators.append(op)
+                new_coeffs.append(c)
+
+        operators = new_operators
+        coefficients = new_coeffs
```

Comment:
For (small) performance reasons, it would be better to store the coefficients as a jnp array
I can't write the suggestion here, but you should put it below the `self._dtype=dtype` line 
`self._coeff = jnp.asarray(coefficients, dtype=dtype)`